src/analyzer/protocol/rdp/rdp-protocol.pac

```diff
@@ -74,18 +74,27 @@ type Data_Block = record {
 # Client X.224
 ######################################################################
 
+type RDP_Connect_Request_cookie = record {
+	cookie_value:          RE/[^\x0d]*/;
+	cookie_terminator:     RE/\x0d\x0a/;
+} &byteorder=littleendian;
+
 type Connect_Request(cotp: COTP) = record {
 	destination_reference: uint16;
 	source_reference:      uint16;
 	flow_control:          uint8;
-	cookie_mstshash:       RE/Cookie: mstshash\=/;
-	cookie_value:          RE/[^\x0d]*/;
-	cookie_terminator:     RE/\x0d\x0a/;
+	cookie_mstshash:       RE/(Cookie: mstshash\=){0,1}/;
```

Comment:
Does that work in BinPac?

```suggestion
	cookie_mstshash:       RE/(Cookie: mstshash\=)?/;
```

Reply:
It might, let me check. I tried a different approach first that didn't - which is why I ended up with `{0,1}`.